src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,855 @@
+                return _registerCache[(byte)Register.Mode];
+            }
+        }
+
+        /// <summary>
+        /// The value of the current Configuration register
+        /// </summary>
+        public uint Config
+        {
+            get
+            {
+                _registerCache[(byte)Register.Configuration] = GetRegisterValue(Register.Configuration);
+                return _registerCache[(byte)Register.Configuration];
+            }
+        }
+
+        /// <summary>
+        /// Power-down mode. In power-down mode, all AD7193 circuitry, except the bridge power-down switch, is powered down. The bridge power-down switch remains active because the user may need to power up the sensor prior to powering up the AD7193 for settling reasons. The external crystal, if selected, remains active.
+        /// </summary>
+        public bool IsPowerDown
+        {
+            get
+            {
+                GetRegisterValue(Register.Mode);
+                return (GetRegisterBits(Register.Mode, BitMask.ModeMD) == 0b011);
+            }
+        }
+
+        /// <summary>
+        /// Ready bit for the ADC. This bit is set when data is written to the ADC data register.
+        /// </summary>
+        public bool IsReady
+        {
+            get
+            {
+                GetRegisterValue(Register.Status);
+                return (GetRegisterBits(Register.Mode, BitMask.StatusRDY) != 0b1);
+            }
+        }
+
+        /// <summary>
+        /// True is the ADC has errors
+        /// </summary>
+        public bool HasErrors
+        {
+            get
+            {
+                uint status = GetRegisterValue(Register.Status);
+
+                return (status & 0b0100_0000) == 0b0100_0000;
+            }
+        }
+
+        private bool _continuousRead = false;
+
+        /// <summary>
+        /// Is the ADC in continuous conversion mode
+        /// </summary>
+        public bool ContinuousRead
+        {
+            get
+            {
+                return _continuousRead;
+            }
+
+            set
+            {
+                if (value)
+                {
+                    SetRegisterValue(Register.Communications, 0b0101_1100);
+                }
+                else
+                {
+                    SetRegisterValue(Register.Communications, 0b0101_1000);
+                }
+
+                _continuousRead = value;
+            }
+        }
+
+        /// <summary>
+        /// Enables or disables DAT_STA Bit (appends status register to data register when reading)
+        /// </summary>
+        public bool AppendStatusRegisterToData
+        {
+            get
+            {
+                return (GetRegisterBits(Register.Mode, BitMask.ModeDAT_STA) == 1);
+            }
+
+            set
+            {
+                _registerCache[(byte)Register.Mode] = SetRegisterBits(Register.Mode, BitMask.ModeDAT_STA, value ? 1U : 0);
```

Comment:
can the caching happen directly in SetRegisterBits? how do you know cache is uninitialized vs initialized with the right value?

Reply:
The cache is initialized with the initial values according to the specification of AD7193.
You're right, the caching should happen in the SetRegisterBits().

Reply:
I moved the cache updater logic into the SetRegisterValue() method, which is also called by the SetRegisterBits() method.